### ISSUE TITLE
[clang-tidy] False positive 'bugprone-derived-method-shadowing-base-method' with template method

### ISSUE THREAD

**Issue by sjoubert** [llvm/llvm-project#185741]
Branching off from #185636

See https://godbolt.org/z/qrM5nWEbs

```cpp
class Base
{
public:
    virtual void work();
};
class Derived: public Base
{
public:
    void work() override;
    
    template <typename T>
    void work();
};
```
```
<source>:13:5: warning: 'Derived::work' shadows method with the same name in class 'Base' [bugprone-derived-method-shadowing-base-method]
   13 |     void work();
      |     ^
<source>:4:5: note: previous definition of 'work' is here
    4 |     virtual void work();
      |     ^
1 warning generated.
```
The documentation specifies `Only checks public, non-templated methods.` but I'm not sure if it applies to the shadowing method, the shadowed one or both.

Without the non-template override of `Derived::work` I'd agree with the warning but given the override, I don't think it shadows anything. Am I missing something about the check reasoning or is it indeed a false positive?

**Comment by vbvictor** [comment]
CC @t-a-james as author
